### PR TITLE
Separate SGX error handling into a new file

### PR DESCRIPTION
**Pull request by manju956**
This change is done to avoid SGX dependency on any module which just needs
to use general error handling functionalities provided by error.h

Signed-off-by: manju956 <manjunath.a.c@intel.com>

### CODE REVIEW

File: tc/sgx/trusted_worker_manager/enclave/base_enclave.cpp [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #include "auto_handle_sgx.h"
 
 #include "error.h"
+#include "avalon_sgx_error.h"
 #include "tcf_error.h"
```
Comment (manojgop) [inline tc/sgx/trusted_worker_manager/enclave/base_enclave.cpp]: We can combine error.h and tcf_error.h into avalon_error.h. May be in another PR ?

Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave/base_enclave.cpp]: Yes, i will combine them in another PR.

File: tc/sgx/common/packages/base64/CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2019 Intel Corporation
+# Copyright 2020 Intel Corporation
```
Comment (manojgop) [inline tc/sgx/common/packages/base64/CMakeLists.txt]: Not a related change for SGX error handling. But fine as it's a minor change updating copyright year.

File: tc/sgx/common/error.h [hunk 3]
```diff
@@ -23,28 +23,6 @@
```
Comment (manojgop) [inline tc/sgx/common/error.h]: sgx_error.h inclusion has to be removed

Comment (manju956) [reply tc/sgx/common/error.h]: Done
